thing-url-adapter.js

```diff
@@ -103,39 +104,52 @@ class ThingURLAdapter extends Adapter {
   constructor(addonManager, packageName) {
     super(addonManager, 'ThingURLPlugin', packageName);
     addonManager.addAdapter(this);
-    this.knownUrls = new Set();
+    this.knownUrls = {};
   }
 
-  loadThing(url) {
+  async loadThing(url) {
     url = url.replace(/\/$/, '');
 
-    if (this.knownUrls.has(url)) {
+    let res;
+    try {
+      res = await fetch(url, {headers: {Accept: 'application/json'}});
+    } catch(e) {
+      console.log('Failed to connect to', url, ':', e);
```

Comment:
Should return here or something instead of just logging.

Reply:
Good catch, thanks!